File: src/main/java/com/example/examplemod/ExampleModBlockStateProvider.java
package com.example.examplemod;

import net.minecraft.block.Block;
import net.minecraft.data.DataGenerator;
import net.minecraftforge.client.model.generators.BlockStateProvider;
import net.minecraftforge.client.model.generators.ModelFile;
import net.minecraftforge.common.data.ExistingFileHelper;

public class ExampleModBlockStateProvider extends BlockStateProvider {
    public ExampleModBlockStateProvider(DataGenerator gen, String modId, ExistingFileHelper exFileHelper) {
        super(gen, modId, exFileHelper);
    }

    @Override
    protected void registerStatesAndModels() {
        simpleBlockWithItem(Blocks.TITANIUM_BLOCK.get());
    }

    private void simpleBlockWithItem(Block block) {
        ModelFile model = cubeAll(block);
        simpleBlock(block, model);
        simpleBlockItem(block, model);
    }
}

File: src/main/java/com/example/examplemod/ExampleModBlockStateProvider.kt
package com.example.examplemod

import net.minecraft.block.Block
import net.minecraft.data.DataGenerator
import net.minecraftforge.client.model.generators.BlockStateProvider
import net.minecraftforge.common.data.ExistingFileHelper


class ExampleModBlockStateProvider(gen: DataGenerator, modId: String, existingFileHelper: ExistingFileHelper) : BlockStateProvider(gen,modId,existingFileHelper){
    override fun registerStatesAndModels() {
        simpleBlockWithItem(ExampleBlocks.TITANIUM_BLOCK.get())
    }

    private fun simpleBlockWithItem(block: Block) {
        val model = cubeAll(block)
        simpleBlock(block, model)
        simpleBlockItem(block, model)
    }
}
